ColorzCore/Parser/Macros/AddToPool.cs

```diff
@@ -0,0 +1,56 @@
+﻿using System;
+using System.Collections.Generic;
+using ColorzCore.Lexer;
+
+namespace ColorzCore.Parser.Macros
+{
+    class AddToPool : BuiltInMacro
+    {
+        /*
+         * Macro Usage:
+         * AddToPool(tokens...): adds token to pool, and expands to label name referring to those tokens
+         * AddToPool(tokens..., alignment): adds token to pool and make sure pooled tokens are aligned given alignment        
+         */
+
+        public EAParser ParentParser { get; private set; }
+
+        public AddToPool(EAParser parent)
+        {
+            ParentParser = parent;
+        }
+
+        public override IEnumerable<Token> ApplyMacro(Token head, IList<IList<Token>> parameters)
+        {
+            List<Token> line = new List<Token>(6 + parameters[0].Count);
+
+            string labelName = ParentParser.MakePoolLabelName();
```

Comment:
To not overload the role of EAParser doing everything (which it is dangerously close to), let's try to have this as a static method here.